packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import memoize from 'memize';
+import { kebabCase, startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useMemo, Children, cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import PanelColorSettings from '../panel-color-settings';
+import ContrastChecker from '../contrast-checker';
+import InspectorControls from '../inspector-controls';
+import { useBlockEditContext } from '../block-edit';
+
+const ColorPanel = ( {
+	title,
+	colorSettings,
+	colorPanelProps,
+	contrastCheckerProps,
+	components,
+	panelChildren,
+} ) => (
+	<PanelColorSettings
+		title={ title }
+		initialOpen={ false }
+		colorSettings={ colorSettings }
+		{ ...colorPanelProps }
+	>
+		{ contrastCheckerProps &&
+			components.map( ( Component ) => (
+				<ContrastChecker
+					key={ Component.displayName }
+					textColor={ Component.color }
+					{ ...contrastCheckerProps }
+				/>
+			) ) }
+		{ typeof panelChildren === 'function' ?
+			panelChildren( components ) :
+			panelChildren }
+	</PanelColorSettings>
+);
+const InspectorControlsColorPanel = ( props ) => (
+	<InspectorControls>
+		<ColorPanel { ...props } />
+	</InspectorControls>
+);
+
+export default function __experimentalUseColors(
+	colorConfigs,
+	{
+		panelTitle = __( 'Color Settings' ),
+		colorPanelProps,
+		contrastCheckerProps,
+		panelChildren,
+	} = {
+		panelTitle: __( 'Color Settings' ),
+	},
+	deps = []
+) {
+	const { attributes, setAttributes } = useBlockEditContext();
+
+	const createComponent = useMemo(
+		() =>
+			memoize(
+				( attribute, color ) => ( { children } ) =>
+					// Clone children, setting the style attribute from the color configuration,
+					// if not already set explicitly through props.
+					Children.map( children, ( child ) =>
+						cloneElement( child, {
+							className: color ?
+								`${ child.props.className } has-${ kebabCase( attribute ) }` :
+								child.props.className,
+							style: { [ attribute ]: color, ...child.props.style },
+						} )
+					),
+				{ maxSize: colorConfigs.length }
+			),
+		[ colorConfigs.length ]
+	);
+	const createSetColor = useMemo(
```

Comment:
[`useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback) is likely more adequate here and in similar places. Why `useMemo`?

Reply:
To avoid calling `memoize` unnecessarily.